tests/unit/mocks/module_mock.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+
+# (c) 2023, Denys Denysyev (ddenysyev@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING.GPLv3 or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from unittest.mock import MagicMock
+
+
+class ModuleSucceeded(Exception):
+    pass
+
+
+class ModuleFailed(Exception):
+    pass
+
+
+class ModuleMock(MagicMock):
+    def __init__(self, params, check_mode=False, exit_json=None, fail_json=None):
+        super(ModuleMock, self).__init__()
+
+        self.params = params
+        self.check_mode = check_mode
+
+        if not exit_json:
+            self.exit_json = MagicMock()
+        else:
+            self.exit_json = exit_json
+
+        if not fail_json:
+            self.fail_json = MagicMock()
+        else:
+            self.fail_json = fail_json
```

Comment:
This is the same as setting the parameters as default
```
def __init__(self, params, check_mode=False, exit_json=MagicMock(), fail_json=MagicMock()):
```
Even better, it will work for `False` or `0` or `""` parameters too (now you would rewrite them by MagicMock).

Reply:
> No, if you do like that, exit_json will be referencing to one object MagicMock(). So all tests will use the same mock object

Good point, I didn't know that.

But still, you should change the code. Imagine `exit_json` has value `0` or `""` or `False` or `[]` or `{}`, in all cases, `if not exit_json` would equal to `True`, thus you would be replacing the original value with `MagicMock()`.

Unless it is an intention, I suggest using:
```
if exit_json is not None:
```
as it is not ambiguous and will be triggered only for `None` value.

Reply:
The same for `fail_json`